components/library/libvpx/Makefile

```diff
@@ -11,45 +11,174 @@
 
 #
 # Copyright 2015 Alexander Pyhalov
+# Copyright 2016 Jim Klimov
 #
 
 include ../../../make-rules/shared-macros.mk
 
-COMPONENT_NAME= libvpx
-COMPONENT_VERSION= 1.4.0
-COMPONENT_SUMMARY= The VP8 Codec SDK
-COMPONENT_SRC= $(COMPONENT_NAME)-$(COMPONENT_VERSION)
-COMPONENT_ARCHIVE= $(COMPONENT_SRC).tar.gz
-COMPONENT_ARCHIVE_HASH= \
-  sha256:eca30ea7fae954286c9fe9de9d377128f36b56ea6b8691427783b20c67bcfc13
-COMPONENT_ARCHIVE_URL= \
-  https://github.com/webmproject/libvpx/archive/v$(COMPONENT_VERSION).tar.gz
-COMPONENT_PROJECT_URL = http://www.webmproject.org/
-COMPONENT_LICENSE = BSD
-COMPONENT_LICENSE_FILE = LICENSE
-COMPONENT_CLASSIFICATION = System/Multimedia Libraries
-COMPONENT_FMRI = library/video/libvpx
-
-include $(WS_TOP)/make-rules/prep.mk
-include $(WS_TOP)/make-rules/configure.mk
-include $(WS_TOP)/make-rules/ips.mk
-
-PATH=/usr/gnu/bin:/usr/bin
+COMPONENT_NAME=		libvpx
+COMPONENT_SUMMARY=	The VP8/VP9 royalty-free Codec SDK
+COMPONENT_PROJECT_URL=	http://www.webmproject.org/
+COMPONENT_LICENSE=	BSD
+COMPONENT_LICENSE_FILE=	LICENSE
+COMPONENT_CLASSIFICATION=	System/Multimedia Libraries
+COMPONENT_FMRI=	library/video/$(COMPONENT_NAME)
+
+# The current version is always _1; the older delivered versions (_2+)
+# are in order of growing numbers starting from the oldest we support
+COMPONENT_VERSION_1=	1.6.0
+COMPONENT_SRC_1=		$(COMPONENT_NAME)-$(COMPONENT_VERSION_1)
+COMPONENT_ARCHIVE_1=	$(COMPONENT_SRC_1).tar.gz
+COMPONENT_ARCHIVE_HASH_1= \
+	sha256:e2fc00c9f60c76f91a1cde16a2356e33a45b76a5a5a1370df65fd57052a4994a
+COMPONENT_ARCHIVE_URL_1= \
+	https://github.com/webmproject/$(COMPONENT_NAME)/archive/v$(COMPONENT_VERSION).tar.gz
+SOURCE_DIR_1=			$(COMPONENT_DIR)/$(COMPONENT_SRC_1)
+
+# Older version for legacy compatibility
+COMPONENT_VERSION_2	= 1.4.0
+COMPONENT_ARCHIVE_HASH_2= \
+	sha256:eca30ea7fae954286c9fe9de9d377128f36b56ea6b8691427783b20c67bcfc13
+COMPONENT_ARCHIVE_URL_2= \
+	https://github.com/webmproject/$(COMPONENT_NAME)/archive/v$(COMPONENT_VERSION_2).tar.gz
+COMPONENT_SRC_2=		$(COMPONENT_NAME)-$(COMPONENT_VERSION_2)
+COMPONENT_ARCHIVE_2=	$(COMPONENT_SRC_2).tar.gz
+SOURCE_DIR_2=			$(COMPONENT_DIR)/$(COMPONENT_SRC_2)
+
+# Another intermediate version
+COMPONENT_VERSION_3=	1.5.0
```

Comment:
I kept it just in case some components (new TB?) would specifically work
with 1.5.0 and are not yet compatible with 1.6.0. Can remove :)

--
Typos courtesy of GMail on my Samsung Android

26.07.2016 20:27 пользователь "Alexander Pyhalov" <notifications@github.com>
написал:

> In components/library/libvpx/Makefile
> <https://github.com/OpenIndiana/oi-userland/pull/2067#discussion_r72308652>
> :
>
> > +COMPONENT_ARCHIVE_URL_1= \
> > +	https://github.com/webmproject/$(COMPONENT_NAME)/archive/v$(COMPONENT_VERSION).tar.gz
> > +SOURCE_DIR_1=			$(COMPONENT_DIR)/$(COMPONENT_SRC_1)
> > +
> > +# Older version for legacy compatibility
> > +COMPONENT_VERSION_2	= 1.4.0
> > +COMPONENT_ARCHIVE_HASH_2= \
> > +	sha256:eca30ea7fae954286c9fe9de9d377128f36b56ea6b8691427783b20c67bcfc13
> > +COMPONENT_ARCHIVE_URL_2= \
> > +	https://github.com/webmproject/$(COMPONENT_NAME)/archive/v$(COMPONENT_VERSION_2).tar.gz
> > +COMPONENT_SRC_2=		$(COMPONENT_NAME)-$(COMPONENT_VERSION_2)
> > +COMPONENT_ARCHIVE_2=	$(COMPONENT_SRC_2).tar.gz
> > +SOURCE_DIR_2=			$(COMPONENT_DIR)/$(COMPONENT_SRC_2)
> > +
> > +# Another intermediate version
> > +COMPONENT_VERSION_3=	1.5.0
>
> We don't need 1.5.0 binary, we have never delivered it
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/OpenIndiana/oi-userland/pull/2067/files/fc74bf1be1058091903ec605cbd9d40e7a55885a#r72308652>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABj5lLQZH0qhhsLZ3kSaXSf0km2YCD9Qks5qZlFwgaJpZM4IpMZj>
> .
>